nornir/plugins/tasks/networking/napalm_cli.py

```diff
@@ -14,6 +14,6 @@ def napalm_cli(task: Task, commands: List[str]) -> Result:
         Result object with the following attributes set:
           * result (``dict``): result of the commands execution
     """
-    device = task.host.get_connection("napalm")
+    device = task.host.get_connection("napalm").connection
```

Comment:
This is strange to me...

I would think `.get_connection("napalm")` should return the napalm connection objection and not the nornir connection object.

In almost every case this is what we want:

```
./plugins/tasks/networking/napalm_validate.py:    device = task.host.get_connection("napalm").connection
./plugins/tasks/networking/napalm_configure.py:    device = task.host.get_connection("napalm").connection
./plugins/tasks/networking/netmiko_file_transfer.py:    net_connect = task.host.get_connection("netmiko").connection
./plugins/tasks/networking/netmiko_send_command.py:    net_connect = task.host.get_connection("netmiko").connection
./plugins/tasks/networking/napalm_get.py:    device = task.host.get_connection("napalm").connection
./plugins/tasks/networking/napalm_cli.py:    device = task.host.get_connection("napalm").connection
./plugins/tasks/networking/netmiko_send_config.py:    net_connect = task.host.get_connection("netmiko").connection
./plugins/tasks/commands/remote_command.py:    connection = task.host.get_connection("paramiko")
./plugins/tasks/files/sftp.py:    client = task.host.get_connection("paramiko").connection
```

The exception being the remote_command Paramiko task for the SSH forwarding agent.  

IMO it should be the other way i.e. get_connection("napalm") returns the napalm connection object then if a task wants the Nornir connection object they should have to do something else special to return the nornir connection object.

The thing I don't like about it is that from a library user perspective it doesn't do what you would expect.

In the SSH agent case, we have inventory information (i.e. the intent to use SSH agent, that we are storing in an alternate config file 'ssh_config' that we are then trying to stuff into Nornir connection state). This doesn't sound like the right way to do this...but would have to dig into it more.

Reply:
I agree with Kirk. I don't really like this repetition.
I propose having:
`def get_connection(self, connection: str) -> Any`
which returns `self.connections[connection].connection`
instead of
`def get_connection(self, connection: str) -> ConnectionPlugin`
and
```
def get_connection_state(self, connection: str) -> Dict[str, Any]:
    if connection not in self.connections:
        raise ValueError(f'Connection {connection} is not open`)
    return self.connections[connection].state
```